perun-core/src/test/java/cz/metacentrum/perun/core/entry/MembersManagerEntryIntegrationTest.java

```diff
@@ -1509,6 +1511,22 @@ public void createSponsoredMemberUsingSeparatedNameMissingFirstNameFail() throws
 		Member sponsoredMember = perun.getMembersManagerBl().createSponsoredMember(sess, createdVo, "dummy", nameOfUser1, "TB", sponsorUser, false);
 	}
 
+	@Test
+	public void setSponsoredMember() throws Exception {
+		System.out.println(CLASS_NAME + "setSponsoredMember");
+		Candidate candidate = new Candidate();
+		candidate.setFirstName("Jan");
+		candidate.setLastName("Novák");
+		User sponsoredUser = perun.getUsersManagerBl().createUser(sess, candidate);
+		Member sponsorMember = setUpSponsor(createdVo);
+		User sponsor = perun.getUsersManagerBl().getUserByMember(sess, sponsorMember);
+		AuthzResolverBlImpl.setRole(sess, sponsor, createdVo, Role.SPONSOR);
+		Member sponsoredMember = perun.getMembersManagerBl().setSponsoredMember(sess, createdVo, sponsoredUser, "dummy", "password", sponsor, true);
+
+		assertNotNull(perun.getMembersManagerBl().getMemberByUser(sess, createdVo, sponsoredUser));
+		assertTrue(sponsoredMember.isSponsored());
```

Comment:
```suggestion
		Member memberFromDb = perun.getMembersManagerBl().getMemberByUser(sess, createdVo, sponsoredUser);
		assertTrue(memberFromDb.isSponsored());
```
@HejdaJakub This is what I meant. And I think that you can actually also omit the `assertNotNull` check since the method should never return null.